### DISCUSSION TITLE
why cant i put different tweens together in a track or sequence?

### DISCUSSION THREAD

**Discussion by astreetsign** [djeedai/bevy_tweening#48] [Q&A]
why cant i put different tweens together in a track or sequence when they have different lenses for example UiPositionLens and TransformRotateZLens?

**Comment by djeedai** [comment]
Because `Animator<T>` is generic on, and therefore animates, a single component type, so it's impossible and wouldn't make sense to put lenses for different components in the same tweenable collection. The `Animator<T>` has to retrieve exactly 1 component using a `Query`, and needs to know the type of that component, and therefore cannot handle multiple ones, let alone of different types. However, since the animator is generic on the component type, you can add one `Animator<Transform>` and one `Animator<Style>` on the same entity, and have one lens on each.

**Comment by djeedai** [comment]
Closing this because the latest version doesn't have `Tracks` anymore and removed generics.
